packages/uniforms-bridge-zod/src/ZodBridge.ts

```diff
@@ -55,15 +55,27 @@ type Option<Value> = {
   value: Value;
 };
 
+type BuildTuple<L extends number, T extends any[] = []> = T['length'] extends L
+  ? T
+  : BuildTuple<L, [...T, any]>;
+
+type Decrement<N extends number> =
+  BuildTuple<N> extends [any, ...infer R] ? R['length'] : never;
+
+type NestedZodEffect<
+  T extends ZodObject<any>,
+  Depth extends number = 20,
+> = Depth extends 0 ? T : ZodEffects<T | NestedZodEffect<T, Decrement<Depth>>>;
+
 export default class ZodBridge<T extends ZodRawShape> extends Bridge {
-  schema: ZodObject<T> | ZodEffects<ZodObject<T>>;
+  schema: ZodObject<T> | NestedZodEffect<ZodObject<T>>;
```

Comment:
```suggestion
  schema: ZodObject<T> | ZodEffects<ZodSchema>
```

We decided we don't want to create a special type with depth limit to handle this. `ZodEffects<ZodSchema>` will allow nested `ZodEffects` type. It will be easier to understand, maintain and we don't want to hardcode specific depth limit.

The downside is that it `ZodSchema` allows any zod type. This is something that we need to accept.

Wrong type is still checked using `invariant` function in the runtime.